### PR TITLE
pkg/kubelet/cm: Remove deprecated sets.String and sets.Int

### PR DESCRIPTION
**Pull request by gjkim42**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

This removes deprecated sets.String and sets.Int
- replace sets.String with sets.Set[string]
- replace sets.Int with sets.Set[int]
- replace sets.NewString with sets.New[string]
- replace sets.NewInt with sets.New[int]
- replace sets.(OLD).List with sets.List(NEW)

This is required for https://github.com/kubernetes/kubernetes/pull/120461
xref: https://github.com/kubernetes/kubernetes/pull/120461#discussion_r1338514856

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by gjkim42** [comment]
/cc @swatisehgal 

This should be merged before https://github.com/kubernetes/kubernetes/pull/120461.

File: pkg/kubelet/cm/devicemanager/manager_test.go [hunk 1]
```diff
@@ -287,7 +287,7 @@ func setupDeviceManager(t *testing.T, devs []*pluginapi.Device, callback monitor
 
 	// test steady state, initialization where sourcesReady, containerMap and containerRunningSet
 	// are relevant will be tested with a different flow
-	err = w.Start(activePods, &sourcesReadyStub{}, containermap.NewContainerMap(), sets.NewString())
+	err = w.Start(activePods, &sourcesReadyStub{}, containermap.NewContainerMap(), sets.New[string]())
```
Comment (gjkim42) [inline pkg/kubelet/cm/devicemanager/manager_test.go:290]: found that this is called by another package...

I'll see if we can change it as well.

**Comment by gjkim42** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/120911/pull-kubernetes-typecheck/1707015055557529600

I guess we may need to change the whole pkg/kubelet ...

**Comment by gjkim42** [comment]
This cleanup PR is required for #120461 which is important-soon.

/priority important-soon

**Review by swatisehgal — APPROVED** [review]
/lgtm

Thanks for this cleanup.

/assign @mrunalp @klueska 
for approval

**Comment by gjkim42** [comment]
> This cleanup PR is required for #120461 which is important-soon.
> 
> /priority important-soon

/remove-priority important-longterm

**Pull request merged**